esphome/components/tuya/tuya.h

```diff
@@ -16,7 +16,7 @@
 namespace esphome {
 namespace tuya {
 
-enum class TuyaDatapointType : uint8_t {
+enum TuyaDatapointType {
```

Comment:
Dont remove the restrictions here
```suggestion
enum class TuyaDatapointType : uint8_t {
```

Reply:
I have to remove them, because of https://github.com/esphome/issues/issues/5984

Reply:
Ah, the solution is to add `is_class=True` to the enum in python. See my latest suggestion